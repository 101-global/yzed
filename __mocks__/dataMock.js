export const model = {
  name: 'YZED',
  brandName: 'YZED',
  imageSlug: 'yzed',
  glbFile: 'https://www.testYZEDfile.com/glbFile.glb',
  usdzFile: 'https://www.testYZEDfile.com/usdzFile.usdz',
  animatedGlbFile: 'https://www.testYZEDfile.com/animatedGlbFile.glb',
  exposure: 1,
};
