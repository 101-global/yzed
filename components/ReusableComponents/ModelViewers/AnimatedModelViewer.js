import React, { useContext, useEffect, useRef, useState } from 'react';

import DownChevron from '../../ReusableComponents/Images/DownChevron';
import PropTypes from 'prop-types';
import { ThemeContext } from '../../../utilities/context/ThemeContext';
import darkBackgroundLoading from '../../../public/yzed-loading-dark-background.svg';
import lightBackgroundLoading from '../../../public/yzed-loading-light-background.svg';

const AnimatedModelViewer = ({ model, topModelDark }) => {
  const modelRef = useRef(null);
  const { darkMode } = useContext(ThemeContext);
  return (
    <>
      <div className='model-viewer-container'>
        <model-viewer
          ref={modelRef}
          src={model.animatedGlbFile}
          alt={model.id}
          loading='lazy'
          camera-controls
          poster={topModelDark || darkMode.value ? darkBackgroundLoading : lightBackgroundLoading}
          exposure={model.exposure}
          interaction-policy='allow-when-focused'
          autoplay>
          {/* <DownChevron topModelDark={topModelDark} /> */}
        </model-viewer>
      </div>
      <style jsx global>{`
        .model-viewer-container {
          height: 100%;
          width: 100%;
        }
        model-viewer {
          margin: 0 auto;
          z-index: 101;
          height: 100%;
          width: 100%;
          --poster-color: transparent;
        }
      `}</style>
    </>
  );
};

export default AnimatedModelViewer;

AnimatedModelViewer.propTypes = {
  model: PropTypes.object.isRequired,
  topModelDark: PropTypes.bool,
};
