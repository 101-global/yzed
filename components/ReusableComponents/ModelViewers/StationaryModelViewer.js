import '@google/model-viewer/dist/model-viewer';

import React, { useContext, useEffect, useRef, useState } from 'react';

import ARActivation from '../Buttons/ARActivation';
import PropTypes from 'prop-types';
import { ThemeContext } from '../../../utilities/context/ThemeContext';
import darkBackgroundLoading from '../../../public/yzed-loading-dark-background.svg';
import dynamic from 'next/dynamic';
import lightBackgroundLoading from '../../../public/yzed-loading-light-background.svg';

const StationaryModelViewer = ({ model }) => {
  const [loading, setLoading] = useState(false);
  const { darkMode } = useContext(ThemeContext);
  const modelRef = useRef(null);

  useEffect(() => {
    if (modelRef.current) {
      modelRef.current.addEventListener('error', (error) => {
        console.log(error);
      });
    }
  }, []);

  return (
    <>
      <div className='model-viewer-container'>
        <model-viewer
          ref={modelRef}
          src={model.glbFile}
          ios-src={model.usdzFile}
          alt={model.id}
          loading='lazy'
          ar-modes='scene-viewer webxr quick-look'
          ar-scale='fixed'
          camera-controls
          ar
          exposure={model.exposure}
          poster={darkMode.value ? darkBackgroundLoading : lightBackgroundLoading}
          interaction-policy='allow-when-focused'>
          <ARActivation />
        </model-viewer>
      </div>
      <style jsx global>{`
        .model-viewer-container {
          height: 100%;
          width: 100%;
        }
        .model-viewer-container model-viewer {
          margin: 0 auto;
          z-index: 101;
          height: 100%;
          width: 100%;
          --poster-color: transparent;
        }
      `}</style>
    </>
  );
};

export default StationaryModelViewer;

StationaryModelViewer.propTypes = {
  model: PropTypes.object.isRequired,
};
