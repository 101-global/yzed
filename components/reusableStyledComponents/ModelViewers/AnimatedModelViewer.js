import React, { useContext, useEffect, useRef, useState } from 'react';

import DownChevron from '../DownChevron';
import PropTypes from 'prop-types';
import darkBackgroundLoading from '../../../public/yzed-loading-dark-background.svg';
import lightBackgroundLoading from '../../../public/yzed-loading-light-background.svg';

const AnimatedModelViewer = ({ model, topModelDark }) => {
  const modelRef = useRef(null);

  return (
    <>
      <div className='model-viewer-container'>
        <model-viewer
          ref={modelRef}
          src={model.animatedGlbFile}
          alt={model.id}
          loading='lazy'
          auto-rotate
          camera-controls
          poster={topModelDark ? darkBackgroundLoading : lightBackgroundLoading}
          exposure={model.exposure}
          interaction-policy='allow-when-focused'
          autoplay>
          <DownChevron topModelDark={topModelDark} />
        </model-viewer>
      </div>
      <style jsx global>{`
        .model-viewer-container {
          height: 100%;
          width: 100%;
        }
        model-viewer {
          margin: 0 auto;
          z-index: 101;
          height: 100%;
          width: 100%;
          --poster-color: transparent;
        }
      `}</style>
    </>
  );
};

export default AnimatedModelViewer;

AnimatedModelViewer.propTypes = {
  model: PropTypes.object.isRequired,
  topModelDark: PropTypes.bool.isRequired,
};
