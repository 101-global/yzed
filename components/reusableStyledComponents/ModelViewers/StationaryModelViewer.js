import React, { useContext, useEffect, useRef, useState } from 'react';

import Head from 'next/head';
import PropTypes from 'prop-types';

const StationaryModelViewer = ({ model }) => {
  const modelRef = useRef(null);
  const [loading, setLoading] = useState(false);

  return (
    <>
      <Head>
        <script
          type='module'
          src='https://unpkg.com/@google/model-viewer/dist/model-viewer.js'></script>
        <script
          noModule
          src='https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js'></script>
      </Head>
      <div className='model-viewer-container'>
        {loading && <h1>Loading</h1>}
        <model-viewer
          ref={modelRef}
          src={model.glbFile}
          ios-src={model.usdzFile}
          alt={model.name}
          preload
          auto-rotate
          camera-controls
          ar
          exposure={0.5}
          poster={'YZED_loadingkopie.png'}
          interaction-policy='allow-when-focused'></model-viewer>
      </div>
      <style jsx global>{`
        .model-viewer-container {
          padding: 20px 0;
          min-height: 700px;
          width: 100%;
        }
        model-viewer {
          width: 100%;
          height: 750px;
          margin: 0 auto;
          --poster-color: transparent;
        }
      `}</style>
    </>
  );
};

export default StationaryModelViewer;

StationaryModelViewer.propTypes = {
  model: PropTypes.object.isRequired,
};
