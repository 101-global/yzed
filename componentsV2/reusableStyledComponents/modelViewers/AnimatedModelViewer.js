import React, { useContext, useEffect, useRef, useState } from 'react';

import Head from 'next/head';
import PropTypes from 'prop-types';
import svg from '../../../public/yzed_logo_outline_gradient_animated.svg';

const AnimatedModelViewer = ({ model }) => {
  const modelRef = useRef(null);

  return (
    <>
      <Head>
        <script
          type='module'
          src='https://unpkg.com/@google/model-viewer/dist/model-viewer.js'></script>
        <script
          nomodule
          src='https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js'></script>
      </Head>
      <div className='model-viewer-container'>
        <model-viewer
          ref={modelRef}
          src={model.animatedGlbFile}
          alt={model.name}
          preload
          auto-rotate
          camera-controls
          poster={svg}
          exposure={0.5}
          interaction-policy='allow-when-focused'
          autoplay></model-viewer>
      </div>
      <style jsx global>{`
        .model-viewer-container {
          height: 100%;
          width: 100%;
        }
        model-viewer {
          margin: 0 auto;

          height: 100%;
          width: 100%;
          --poster-color: transparent;
        }
      `}</style>
    </>
  );
};

export default AnimatedModelViewer;

AnimatedModelViewer.propTypes = {
  model: PropTypes.object.isRequired,
};
