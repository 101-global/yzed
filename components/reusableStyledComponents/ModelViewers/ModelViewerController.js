import React, { useContext, useEffect, useState } from 'react';

import AnimatedModelViewer from './AnimatedModelViewer';
import AnimatedVideoPlayer from './AnimatedVideoPlayer';
import PropTypes from 'prop-types';
import StationaryModelViewer from './StationaryModelViewer';
import SwitchSelector from 'react-switch-selector';
import theme from '../../../utilities/theme';

const ModelViewerController = ({ model, topModelDark, setTopModelDark }) => {
  const [showAnimated, setShowAnimated] = useState(false);
  const [isDesktop, setIsDesktop] = useState(false);
  const [isVideoLoading, setIsVideoLoading] = useState(true);

  const animationOptions = [
    {
      label: 'Stationary',
      value: false,
    },
    {
      label: 'Walking',
      value: true,
    },
  ];

  const lightOptions = [
    { label: 'Light', value: false },
    { label: 'Dark', value: true },
  ];

  const onAnimationChange = (newValue) => {
    setShowAnimated(newValue);
    if (newValue === true) {
      setIsVideoLoading(true);
    } else {
      setIsVideoLoading(false);
    }
  };

  const onLightChange = (newValue) => {
    setTopModelDark(newValue);
  };

  const onLoadedData = () => {
    setIsVideoLoading(false);
  };

  useEffect(() => {
    if (window.innerWidth > 1024) {
      setIsDesktop(true);
    }
  }, []);

  return (
    <>
      <div className='viewer-container flex justify-between items-center'>
        <div className='model-viewer-container'>
          {showAnimated ? (
            model.animatedVideo && !isDesktop ? (
              <AnimatedVideoPlayer
                model={model}
                isVideoLoading={isVideoLoading}
                onLoadedData={onLoadedData}
                topModelDark={topModelDark}
              />
            ) : (
              <AnimatedModelViewer model={model} topModelDark={topModelDark} />
            )
          ) : (
            <StationaryModelViewer model={model} topModelDark={topModelDark} />
          )}
        </div>
        <div className='controls'>
          <div className='toggle-animated-buttons'>
            <SwitchSelector
              onChange={onAnimationChange}
              options={animationOptions}
              initialSelectedIndex={0}
              backgroundColor={theme.colors.black}
              fontColor={theme.colors.white}
              selectedFontColor={theme.colors.black}
              selectedBackgroundColor={theme.colors.white}
              optionBorderRadius={0}
              selectionIndicatorMargin={3}
            />
          </div>
          <div className='toggle-light-buttons'>
            <SwitchSelector
              onChange={onLightChange}
              options={lightOptions}
              initialSelectedIndex={1}
              backgroundColor={theme.colors.black}
              fontColor={theme.colors.white}
              selectedFontColor={theme.colors.black}
              selectedBackgroundColor={theme.colors.white}
              optionBorderRadius={0}
              selectionIndicatorMargin={3}
            />
          </div>
        </div>
      </div>
      <style jsx>{`
        .video-loading-logo {
          height: 90vh;
          z-index: 1000;
        }
        .viewer-container {
          height: 90vh;
          max-width: 80%;
          margin: 0 auto;
        }
        .video-container {
          height: 80vh;
        }
        .controls {
          display: flex;
          flex-direction: column;
          justify-content: space-around;
          position: absolute;
          top: 4.8rem;
          right: 10%;
          z-index: 1000;
          background: transparent;
          color: ${!topModelDark ? theme.colors.black : theme.colors.white};
        }

        .toggle-animated-buttons,
        .toggle-light-buttons {
          height: 40px;
          width: 380px;
          margin-bottom: 20px;
          margin-top: 10px;
        }
        @media (max-width: 768px) {
          .controls {
            position: relative;
            display: block;
            top: 0;
            bottom: 0;
            right: 0;
            left: 0;
            width: 100%;
            padding: 2.5%;
            background: ${topModelDark ? theme.colors.black : theme.colors.white};
          }
          .viewer-container {
            display: flex;
            flex-direction: column;
            max-width: 100%;
          }
          .model-viewer-container {
            min-height: 80vh;
            max-width: 90%;
            margin: 0 auto;
          }
          .toggle-animated-buttons,
          .toggle-light-buttons {
            width: 100%;
            margin-bottom: 0px;
          }
      `}</style>
      <style jsx global>{`
        .react-switch-selector-wrapper {
          border-radius: 0px !important;
          border: 1px solid ${theme.colors.white} !important;
          height: 30px !important;
        }
        @media (max-width: 640px) {
          .react-switch-selector-wrapper {
            max-width: 100% !important;
          }
        }
      `}</style>
    </>
  );
};

export default ModelViewerController;

ModelViewerController.propTypes = {
  model: PropTypes.object.isRequired,
  topModelDark: PropTypes.bool.isRequired,
  setTopModelDark: PropTypes.func.isRequired,
};
