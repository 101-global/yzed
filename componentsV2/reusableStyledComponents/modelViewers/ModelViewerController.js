import React, { useContext, useEffect, useState } from 'react';

import AnimatedModelViewer from './AnimatedModelViewer';
import PropTypes from 'prop-types';
import StationaryModelViewer from './StationaryModelViewer';
import SwitchSelector from 'react-switch-selector';
import theme from '../../../utilities/theme';

const ModelViewerController = ({ model, topModelDark, setTopModelDark }) => {
  const [showAnimated, setShowAnimated] = useState(false);
  const animationOptions = [
    {
      label: 'Stationary',
      value: false,
    },
    {
      label: 'Walking',
      value: true,
    },
  ];

  const lightOptions = [
    { label: 'Light', value: false },
    { label: 'Dark', value: true },
  ];

  const onAnimationChange = (newValue) => {
    setShowAnimated(newValue);
  };

  const onLightChange = (newValue) => {
    setTopModelDark(newValue);
  };

  return (
    <>
      <div className='control-container'>
        <div className='controls-section'>
          <div className='controls'>
            <div className='toggle-animated-buttons'>
              <SwitchSelector
                onChange={onAnimationChange}
                options={animationOptions}
                initialSelectedIndex={0}
                backgroundColor={'#0d0d0d'}
                fontColor={'#fff'}
                selectedFontColor={'#0d0d0d'}
                selectedBackgroundColor={'#fff'}
                optionBorderRadius={0}
                selectionIndicatorMargin={3}
              />
            </div>
            <div className='toggle-light-buttons'>
              <SwitchSelector
                onChange={onLightChange}
                options={lightOptions}
                initialSelectedIndex={1}
                backgroundColor={'#0d0d0d'}
                fontColor={'#fff'}
                selectedFontColor={'#0d0d0d'}
                selectedBackgroundColor={'#fff'}
                optionBorderRadius={0}
                selectionIndicatorMargin={3}
              />
            </div>
          </div>
        </div>

        <div className='viewer-container'>
          {showAnimated ? (
            <AnimatedModelViewer model={model} />
          ) : (
            <StationaryModelViewer model={model} />
          )}
        </div>
      </div>
      <style jsx>{`
        .viewer-container {
          height: 100%;
          width: 2000px;
          max-width: 80%;
          margin: 0 auto;
          display: flex;
          justify-content: center;
          align-items: center;
        }
        .control-container {
          height: 90vh;
          width: 100%;
          display: flex;
        }
        .controls {
          display: inline-block;
          flex-direction: column;
          justify-content: space-around;
          position: absolute;
          top: calc(30px + 3.2rem);
          right: 10vw;
          z-index: 1000;
          min-height: 100px;
          background: transparent;
          color: ${!topModelDark ? theme.colors.black : theme.colors.white};
        }
        .toggle-button-container {
          display: flex;
          flex-direction: row;
          align-items: center;
          justify-content: space-between;
        }

        .toggle-animated-buttons,
        .toggle-light-buttons {
          height: 40px;
          width: 380px;
          margin-bottom: 20px;
          margin-top: 10px;
        }
        @media (max-width: 640px) {
          .controls {
            flex-direction: column;
            justify-content: flex-end;
            bottom: 200px;
            top: calc(100% - 250px);
            margin-bottom: 100px;
            transform: translateY(50%);
            left: 0;
            right: 0;
            max-width: 95%;
          }
          .viewer-container {
            margin-bottom: 100px;
            height: 100%;
            max-width: 90%;
          }
        }
      `}</style>
      <style jsx global>{`
        .react-switch-selector-wrapper {
          border-radius: 0px !important;
          border: 1px solid ${theme.colors.white} !important;
          height: 30px !important;
          margin-bottom: 20px;
          width: 380px;
        }
        @media (max-width: 640px) {
          .react-switch-selector-wrapper {
            max-width: 90%;
            left: calc(5%);
          }
        }
      `}</style>
    </>
  );
};

export default ModelViewerController;

ModelViewerController.propTypes = {
  model: PropTypes.object.isRequired,
  lightMode: PropTypes.bool.isRequired,
  setLightMode: PropTypes.func.isRequired,
};
