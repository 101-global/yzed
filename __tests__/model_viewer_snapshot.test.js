jest.mock('../node_modules/@google/model-viewer/dist/model-viewer', () => () => 'Model Viewer');

import AnimatedModelViewer from '../components/reusableStyledComponents/modelViewers/AnimatedModelViewer';
import ModelViewerController from '../components/reusableStyledComponents/modelViewers/ModelViewerController';
import React from 'react';
import StationaryModelViewer from '../components/reusableStyledComponents/modelViewers/StationaryModelViewer';
import { model } from '../__mocks__/dataMock';
import renderer from 'react-test-renderer';

const mockSetTopModelDark = () => {};

it('renders model viewer controller unchanged', () => {
  const wrapper = renderer
    .create(
      <ModelViewerController
        model={model}
        topModelDark={true}
        setTopModelDark={mockSetTopModelDark}
      />
    )
    .toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('renders stationary model viewer unchanged', () => {
  const wrapper = renderer
    .create(
      <StationaryModelViewer
        model={model}
        topModelDark={true}
        setTopModelDark={mockSetTopModelDark}
      />
    )
    .toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('renders animated model viewer unchanged', () => {
  const wrapper = renderer
    .create(
      <AnimatedModelViewer
        model={model}
        topModelDark={true}
        setTopModelDark={mockSetTopModelDark}
      />
    )
    .toJSON();
  expect(wrapper).toMatchSnapshot();
});
