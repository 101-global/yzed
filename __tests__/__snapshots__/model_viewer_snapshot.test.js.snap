// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders animated model viewer unchanged 1`] = `
<div
  className="jsx-1120965399 model-viewer-container"
>
  <model-viewer
    auto-rotate={true}
    autoplay={true}
    camera-controls={true}
    className="jsx-1120965399"
    exposure={1}
    interaction-policy="allow-when-focused"
    loading="lazy"
    poster="File Sub Mock"
    src="https://www.testYZEDfile.com/animatedGlbFile.glb"
  >
    <svg
      className="jsx-3413298786 down-chevron model-chevron block"
      enableBackground="new 0 0 1000 1000"
      version="1.1"
      viewBox="0 0 1000 1000"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <g
        className="jsx-3413298786"
      >
        <path
          className="jsx-3413298786"
          d="M500,757.8L10,271.5l29.3-29.3L500,706.6l460.7-464.4l29.3,29.3L500,757.8z"
          fill="#fff"
        />
      </g>
    </svg>
  </model-viewer>
</div>
`;

exports[`renders model viewer controller unchanged 1`] = `
<div
  className="jsx-2866622310 jsx-3808981773 viewer-container flex justify-between items-center"
>
  <div
    className="jsx-2866622310 jsx-3808981773 model-viewer-container"
  />
  <div
    className="jsx-2866622310 jsx-3808981773 controls"
  >
    <div
      className="jsx-2866622310 jsx-3808981773 toggle-animated-buttons"
    >
      <div
        className="react-switch-selector-wrapper css-yp4xe5"
      >
        <div
          className="react-switch-selector-option css-15e01fz"
        >
          <label
            className="react-switch-selector-option-label css-14spvco"
            fontSize={14}
            selected={true}
          >
            <input
              checked={true}
              className="css-18exkk3"
              id="rss-option-0"
              onChange={[Function]}
              type="radio"
            />
            Stationary
          </label>
        </div>
        <div
          className="react-switch-selector-option css-15e01fz"
        >
          <label
            className="react-switch-selector-option-label css-1nq8m5d"
            fontSize={14}
            selected={false}
          >
            <input
              checked={false}
              className="css-18exkk3"
              id="rss-option-1"
              onChange={[Function]}
              type="radio"
            />
            Walking
          </label>
        </div>
      </div>
    </div>
    <div
      className="jsx-2866622310 jsx-3808981773 toggle-light-buttons"
    >
      <div
        className="react-switch-selector-wrapper css-1te2sdl"
      >
        <div
          className="react-switch-selector-option css-15e01fz"
        >
          <label
            className="react-switch-selector-option-label css-1nq8m5d"
            fontSize={14}
            selected={false}
          >
            <input
              checked={false}
              className="css-18exkk3"
              id="rss-option-0"
              onChange={[Function]}
              type="radio"
            />
            Light
          </label>
        </div>
        <div
          className="react-switch-selector-option css-15e01fz"
        >
          <label
            className="react-switch-selector-option-label css-14spvco"
            fontSize={14}
            selected={true}
          >
            <input
              checked={true}
              className="css-18exkk3"
              id="rss-option-1"
              onChange={[Function]}
              type="radio"
            />
            Dark
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders stationary model viewer unchanged 1`] = `
<div
  className="jsx-731752813 model-viewer-container"
>
  <model-viewer
    ar={true}
    ar-modes="scene-viewer webxr quick-look"
    ar-scale="fixed"
    auto-rotate={true}
    camera-controls={true}
    className="jsx-731752813"
    exposure={1}
    interaction-policy="allow-when-focused"
    ios-src="https://www.testYZEDfile.com/usdzFile.usdz"
    loading="lazy"
    poster="File Sub Mock"
    src="https://www.testYZEDfile.com/glbFile.glb"
  >
    <svg
      className="jsx-3413298786 down-chevron model-chevron block"
      enableBackground="new 0 0 1000 1000"
      version="1.1"
      viewBox="0 0 1000 1000"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <g
        className="jsx-3413298786"
      >
        <path
          className="jsx-3413298786"
          d="M500,757.8L10,271.5l29.3-29.3L500,706.6l460.7-464.4l29.3,29.3L500,757.8z"
          fill="#fff"
        />
      </g>
    </svg>
  </model-viewer>
</div>
`;
